src/Store.ts

```diff
@@ -2,30 +2,81 @@ import { Evented } from '@dojo/core/Evented';
 import { Patch, PatchOperation } from './state/Patch';
 import { Pointer } from './state/Pointer';
 
+export interface Path<M, T> {
+	path: string;
+	state: M;
+	value: T;
+}
+
+export interface State<M> {
```

Comment:
Do you think we should add some jsdoc for interface? (I know generally that could be improved across the repository also...)